### ISSUE TITLE
No DocBlock with TypeScript in Vue

### ISSUE THREAD

**Issue by jonny7** [jeremyvii/vs-docblockr#45]
**Is your feature request related to a problem? Please describe.**
Hi, thanks for this, for me. I don't seem to be able to get a doc block if the code is inside a .vue file. I can however inside either a .ts or .js file. Just wondered if that is the expected behaviour

**Describe the solution you'd like**
docblocks within .vue extensions 

**Describe alternatives you've considered**
none, but might just put it in a ts file and copy it back over

:)

**Comment by jeremyvii** [comment]
Hey, thanks for using the extension. The problem is that docblocks are generated based on the current language ID of the language being used. It may be as easy as assigning TypeScript to the Vue language ID, but I am unsure of what side effects this may have.

It would be a nice enhancement!

**Comment by jonny7** [comment]
Yeah, I know it's not necessarily that easy. SonarQube have had the ability to ready typescript within .vue files on their list of features for a while now. I'm just putting in a .ts file for the moment and importing into the .vue file, so doc block and sonar works. So that's cool :)

**Comment by jeremyvii** [comment]
It might be a bit before I can look into this. If you're interesting in attempting to implement this you can try to add the TypeScript parser for the Vue language [here](/jeremyvii/vs-docblockr/blob/master/src/extension.ts#L35).

This file, [extension.ts](/jeremyvii/vs-docblockr/blob/master/src/extension.ts), loads the parser onto the language ID.
``` typescript
  const langList = {
    java: Java,
    javascript: JavaScript,
    php: PHP,
    scss: Scss,
    typescript: TypeScript,
  };
```

The [package.json](/jeremyvii/vs-docblockr/blob/master/package.json#L18) activates the language.

Then it would just need to be tested to see if the various snippets in the [tests](/jeremyvii/vs-docblockr/blob/master/test/languages/typescript.test.ts), work in the Vue language.

**Comment by jeremyvii** [comment]
TypeScript support in Vue.js will be released in [0.6.0](https://github.com/jeremyvii/vs-docblockr/releases/tag/0.6.0).
